thunder/torch/__init__.py

```diff
@@ -85,6 +85,24 @@
 _inplace_to_out_of_place: dict[Callable, tuple[Callable, int]] = {}
 
 
+# Helpers for factory functions to get default dtypes.
+def get_default_dtype():
+    # `thunder.jit` will create cache info and stash the default dtype
+    # observed at the beginning of jitting.
+    cache_info = thunder._get_cache_info()
```

Comment:
This won't work with the deprecated `thunder.compile` (which is why I had to update `executor.make_callable_with_info(fn)` to `executor.make_callable(fn)`). But I assume that is ok.

Reply:
if we could remove make_callable_with_info, it'd be perfect. :)

Reply:
Updating to `make_callable` worked for the failing test. Resolving the comment :)